flake.nix

```diff
@@ -136,6 +136,11 @@
             runTests = true;
           });
 
+          hopr-test-nightly = rust-builder-local-nightly.callPackage ./nix/rust-package.nix (hoprdBuildArgs // {
```

Comment:
Why is the nightly needed?

Reply:
@Teebor-Choka if a thread panics in unit tests, it does not kill the test's main thread and fail the test. Due to this reason we had several failing tests which did not manifest. 

Nightly makes the tests fail also if a thread panics. 

We should be more diligent and await and test all join handles of tasks spun in tests.